packages/flame/src/hooks/useEventListener.ts

```diff
@@ -5,21 +5,18 @@ type CallbackEvent<T> = (event: T) => void;
 export function useEventListener<T extends Event>(
   eventKey: string,
   callback: CallbackEvent<T>,
-  target?: EventTarget,
+  target: EventTarget = document,
 ) {
   React.useEffect(() => {
     const captureEvent: CallbackEvent<T> = event => {
       callback(event);
     };
 
-    if (target || document) {
-      (target || document).addEventListener(eventKey, captureEvent as any);
+    if (target) {
```

Comment:
Since `target` is mandatory and is an `EventTarget`, do we even need to have this condition?

Reply:
The `if` statement should defs stay and the type nullable as per my other comment, the refs could become null if the element is removed from the dom or during the initial render where the refs is still null.

Reply:
Actually on SSR `document` will not exist at all (depending on your setup).  So to be safe we need a `if(document !== 'undefined')`

Reply:
Also, the typing will always look funny here because `document` itself is supposed to exist and have a value according to TS definitions.

Reply:
Yeah.. I'd leave the `target?: EventTarget` honestly. It's clearer and TS will complain if we don't do the check.

Reply:
Nvm, I saw the updated code. Looks better to me 👍